sklearn/tree/_criterion.pyx

```diff
@@ -1096,45 +949,41 @@ cdef class FriedmanMSE(MSE):
         The absolute impurity improvement is only computed by the
         impurity_improvement method once the best split has been found.
         """
-        cdef SIZE_t n_outputs = self.n_outputs
-        cdef double* sum_left = self.sum_left
-        cdef double* sum_right = self.sum_right
-        cdef double weighted_n_left = self.weighted_n_left
-        cdef double weighted_n_right = self.weighted_n_right
+
+        cdef double* node_sum_left = self.node_sum_left
+        cdef double* node_sum_right = self.node_sum_right
+
         cdef double total_sum_left = 0.0
         cdef double total_sum_right = 0.0
 
-        cdef SIZE_t k = 0
+        cdef SIZE_t k
         cdef double diff = 0.0
 
-        for k in range(n_outputs):
-            total_sum_left += sum_left[k]
-            total_sum_right += sum_right[k]
+        for k in range(self.n_outputs):
+            total_sum_left += node_sum_left[k]
+            total_sum_right += node_sum_right[k]
 
-        diff = (weighted_n_right * total_sum_left -
-                weighted_n_left * total_sum_right)
+        diff = (self.weighted_n_right * total_sum_left -
+                self.weighted_n_left * total_sum_right)
 
-        return diff * diff / (weighted_n_left * weighted_n_right)
+        return diff * diff / (self.weighted_n_left * self.weighted_n_right)
 
     cdef double impurity_improvement(self, double impurity) nogil:
-        cdef SIZE_t n_outputs = self.n_outputs
-        cdef double* sum_left = self.sum_left
-        cdef double* sum_right = self.sum_right
-        cdef double weighted_n_left = self.weighted_n_left
-        cdef double weighted_n_right = self.weighted_n_right
-        cdef double weighted_n_node_samples = self.weighted_n_node_samples
+        cdef double* node_sum_left = self.node_sum_left
+        cdef double* node_sum_right = self.node_sum_right
+
         cdef double total_sum_left = 0.0
         cdef double total_sum_right = 0.0
 
-        cdef SIZE_t k = 0
+        cdef SIZE_t k
         cdef double diff = 0.0
 
-        for k in range(n_outputs):
-            total_sum_left += sum_left[k]
-            total_sum_right += sum_right[k]
+        for k in range(self.n_outputs):
+            total_sum_left += node_sum_left[k]
+            total_sum_right += node_sum_right[k]
 
-        diff = (weighted_n_right * total_sum_left -
-                weighted_n_left * total_sum_right) / n_outputs
+        diff = (self.weighted_n_right * total_sum_left -
+                self.weighted_n_left * total_sum_right) / self.n_outputs
 
-        return (diff * diff /
-                (weighted_n_left * weighted_n_right * weighted_n_node_samples))
\ No newline at end of file
+        return (diff * diff / (self.weighted_n_left * self.weighted_n_right * 
+                self.weighted_n_node_samples))
```

Comment:
Many older tools may misbehave if the last line of a text file is not terminated by a line return. (Because otherwise the last line of context contains EOF and may be ignored)